src/components/contenteditable/contenteditable.component.ts

```diff
@@ -0,0 +1,45 @@
+import {Directive, ElementRef, EventEmitter, Input, OnChanges, Output, Renderer } from '@angular/core';
+
+const KEY_ENTER = 13;
+
+@Directive({
+  selector: '[contenteditableModel]',
+  host: {
+    '(blur)': 'onBlur()',
+    '(keypress)': 'onKeypress($event)'
+  }
+})
+export class ContenteditableModel implements OnChanges {
+  @Input('contenteditableModel') model: any;
```

Comment:
I believe any should be there as it can be anything from an object to string etc.